src/main/java/com/softserveinc/ita/pageobjects/SearchResultPage.java

```diff
@@ -1,19 +1,67 @@
 package com.softserveinc.ita.pageobjects;
 
+import static com.codeborne.selenide.CollectionCondition.sizeGreaterThan;
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$$x;
+import static com.codeborne.selenide.Selenide.$x;
+import static java.lang.String.format;
+import static java.time.Duration.ofSeconds;
 import com.codeborne.selenide.SelenideElement;
 import io.qameta.allure.Step;
 
 import java.util.List;
 
-import static com.codeborne.selenide.CollectionCondition.sizeGreaterThan;
 import static com.codeborne.selenide.CollectionCondition.sizeNotEqual;
-import static com.codeborne.selenide.Condition.visible;
 import static com.codeborne.selenide.Selenide.$;
-import static com.codeborne.selenide.Selenide.$$x;
-import static java.time.Duration.ofSeconds;
 import static java.util.stream.Collectors.toList;
 
-public class SearchResultPage extends BasePage<SearchResultPage> {
+public class SearchResultPage extends  BasePage<SearchResultPage> {
+
+    private final String subCategorySelector =  "//rz-widget-list//ul/li//a[contains(@Class , 'tile-cats__heading')]";
+
+    public SearchResultPage openSubCategoryByOrderNumber(int orderNumber ) {
```

Comment:
selectRandomSubCategoryByNumber

also here not the wright place for this method probably it should be in CategoriesPage